pkg/reconciler/shared/tektonconfig/tektonconfig.go

```diff
@@ -138,3 +154,45 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tc *v1alpha1.TektonConfi
 
 	return nil
 }
+
+func (r *Reconciler) createOperatorVersionConfigMap(tc *v1alpha1.TektonConfig) error {
+	koDataDir := os.Getenv(common.KoEnvKey)
+	operatorDir := filepath.Join(koDataDir, "info")
+
+	if err := common.AppendManifest(&r.manifest, operatorDir); err != nil {
+		return err
+	}
+
+	manifest, err := r.manifest.Transform(
+		mf.InjectNamespace(tc.GetSpec().GetTargetNamespace()),
+	)
+	if err != nil {
+		return err
+	}
+
+	if err = manifest.Apply(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Reconciler) ensureTargetNamespaceExists(ctx context.Context, tc *v1alpha1.TektonConfig) error {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: tc.GetSpec().GetTargetNamespace(),
+			Labels: map[string]string{
+				"operator.tekton.dev/targetNamespace": tc.GetSpec().GetTargetNamespace(),
+			},
+		},
+	}
+	_, err := r.kubeClientSet.CoreV1().Namespaces().Get(ctx, tc.GetSpec().GetTargetNamespace(), metav1.GetOptions{})
```

Comment:
here we have to handle the secanario where the `spec.targetNamespace` being changed by a user.

so ideally, the flow should be, 
1. query (list) namespaces using a label (eg: operator.tekton.dev/targetNamespace: true)
2. if list is non-empty, check if any of the namespaces has name == spec.targetNamespace 
     2.a. if for any namespace in the list name != spec.targetNamespace, then delete it
3. if spec.targetNamespace exists (in the list) then return nil
4. else create spec.targetNamespace